### PR TITLE
cambricon: fix layernorm, improve groupnorm

### PR DESCRIPTION
**Pull request by tututuski**
For groupnorm, add a constant （DEAL_NUM） to mark the number of elements processed at a time. It can also be set to 0, which will make the behavior the same as before the merge.

For layernorm, fixed the problem that BLOCK_ROW_SIZE can only be 1 (taking 2/4/8... will cause calculation errors). After the fix, BLOCK_ROW_SIZE can be 1/2/4/8/...

### CODE REVIEW

File: src/flag_gems/ops/layernorm.py [hunk 1]
```diff
@@ -125,15 +125,15 @@ def layer_norm_backward_kernel(
         mask = row_mask and col_mask
         dy = tl.load(dY + cols[None, :], mask).to(tl.float32)
         x = tl.load(X + cols[None, :], mask).to(tl.float32)
-        x = tl.where(col_mask, x - mean, 0.0)
+        x = tl.where(mask, x - mean, 0.0)
```
Comment (sethbrin) [inline src/flag_gems/ops/layernorm.py:128]: 这个是不是原生 layernorm的问题，如果是的话，可以提一个PR直接到main分支哈

File: src/flag_gems/ops/groupnorm.py [hunk 2]
```diff
@@ -130,33 +138,46 @@ def weight_bias_backward_kernel(
     HW,
     BLOCK_N: tl.constexpr,
     BLOCK_HW: tl.constexpr,
+    DEAL_NUM: tl.constexpr,
 ):
```
Comment (sethbrin) [inline src/flag_gems/ops/groupnorm.py]: 这个是不是直接改调用侧代码呢？直接引入一层循环来处理`BLOCK_HW`?

**Review by sethbrin — COMMENTED** [review]
others, LGTM

**Comment by tututuski** [comment]
放弃 groupnorm的改动， layernorm 的修改直接合入到master

File: src/flag_gems/ops/layernorm.py (hunk 1 above)
Comment (tongxin) [reply src/flag_gems/ops/layernorm.py:128]: Just saw it.

**Pull request closed without merging**